### DISCUSSION TITLE
.mat or .nrrd files

### DISCUSSION THREAD

**Discussion by joshua-crockett** [SCIInstitute/Seg3D#409] [Q&A]
Hi all!
Can Seg3D read .mat files or do the images need to be .nrrd files? And if I need .nrrd files, do you know a way I can convert a .mat file to a .nrrd?
Thanks!

**Comment by jessdtate** [comment]
Seg3D does support a matlab file format, as well as other file formats, including: dicom, nrrd, and tiff.  If you are unable to load your matlab data into Seg3D, you can try altering the matlab file format.  Here are some example files with the structure needed to read into Seg3D.
[matlab_files.zip](https://github.com/SCIInstitute/Seg3D/files/8848351/matlab_files.zip)

Alternatively, nrrd, dicom, and other files can be saved from Matlab and Python.  There is a SCIRun FAQ explaining how to export nrrd files from matlab, but there are existing libraries for many of these formats that should work. 

The main thing to remember when making files to read into Seg3D is that it expects 3D structured hexahedral volumes and that it assumes the data are either a series of label masks, or a grayscale image volume.  Seg3D does not currently support time-dependent or mullti-channel explicitly, but each channel or time-frame can be loaded as an individual volumes.
